Automattic/wp-calypso
# Activity Log: Get display rules from real API endpoint

Resolves `1200412004370260-as-1200706668146547`.

#### Changes proposed in this Pull Request

* Update the data-getter endpoint for Activity Log display rules to match the endpoint in `D65474-code`.
* Update application state logic to accommodate the shape of the API response in `D65474-code`.

#### Testing instructions

For extra context, check out #54900 and #55298.

##### Prerequisites

1. Create or select a connected Jetpack site to use for testing this PR.
2. Add a Backup or Security Tier 1 subscription to your site via the Store Admin.
3. Enable the `activity-log/display-rules` feature flag, either by modifying `development.json`/`jetpack-cloud-development.json` or by adding `flags=activity-log/display-rules` to your browser URL's query string.
4. Open your browser's development tools to the Network panel to monitor XHR requests.

##### New behavior

5. Visit the Activity Log page in Calypso (Blue or Green).
6. Verify you see an outgoing request to `/wpcom/v2/sites/<site id>/rewind/policies`, and that it returns a success response.
7. Verify that if you added a Tier 1 subscription, the `activity_log_limit_days` field has a value of `30`; or, if you selected a Tier 2 subscription, it has a value of `365`.
8. In your browser's console, execute `state.rewind[ state.ui.selectedSiteId ].policies` and examine the result:
  * the `requestStatus` field should equal `success`; and
  * the `activityLogLimitDays` field should match the value of `activity_log_limit_days` from the network response in step 8.
9. Test that on sites with a Tier 1 subscription and more than 30 days of Activity history, you see a notice at the 30-day mark with instructions on how to upgrade your storage (see #54900, #55298).

##### Existing behavior

If you'd like to test for regressions:

10. Remove the Tier 1/Tier 2 subscription from your site. Wait up to 5 minutes for back-end caches to update.
11. Refresh the Activity Log page.
12. Verify you see a successful network request like in step 7, albeit with no data for `activity_log_limit_days`.
13. Verify the application state now reflects an `activityLogLimitDays` value of `null`.
14. Verify there are no unexpected errors in the browser console.
